python/pynq/mmio.py

```diff
@@ -96,16 +97,18 @@ def __init__(self, base_addr, length = 4, debug = False):
         self.length = length
         
         self.debug = debug
-        self._debug('MMIO(address, size) = ({0}, {1} bytes).'.\
-                    format(hex(self.base_addr), hex(self.length)))
+        self._debug('MMIO(address, size) = ({0:x}, {1:x} bytes).',
+                    self.base_addr,self.length)
                 
         # Open file and mmap
-        f = os.open(general_const.MMIO_FILE_NAME, os.O_RDWR | os.O_SYNC)
-        self.mem = mmap.mmap(f, (self.length + self.virt_offset), 
+        self.f = os.open(general_const.MMIO_FILE_NAME, os.O_RDWR | os.O_SYNC)
+        self.mem = mmap.mmap(self.f, (self.length + self.virt_offset),
```

Comment:
Notice that self.f is only used once. Is it really necessary to keep this as an attribute?